File: entities/src/main/java/org/wcl/mfn/entities/contract/calculator/SuggestedContract.java
package org.wcl.mfn.entities.contract.calculator;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Size;

import java.util.List;

public record SuggestedContract(int years, int totalSalary, int bonusPerYear, int totalRemuneration,
                                @Size(min = 2, max = 6) @Valid List<SuggestedYearlyContract> yearlyFigures) {
}

File: entities/src/main/kotlin/org/wcl/mfn/entities/contract/calculator/SuggestedContract.kt
package org.wcl.mfn.entities.contract.calculator

import jakarta.validation.Valid
import jakarta.validation.constraints.Size

@JvmRecord
data class SuggestedContract(
    val years: Int,
    val totalSalary: Int,
    val bonusPerYear: Int,
    val totalRemuneration: Int,
    val yearlyFigures: @Size(min = 2, max = 6) @Valid List<SuggestedYearlyContract>
)

